js/views/settingsVw.js

```diff
@@ -0,0 +1,290 @@
+
+        var submit = function(img_hash) {
+
+            if(img_hash) {
+                profileFormData.append("avatar",img_hash);
+            }
+
+            $.ajax({
+                type: "POST",
+                url: server + "settings",
+                contentType: false,
+                processData: false,
+                data: settingsFormData,
+                success: function(data) {
+                    if(JSON.parse(data).success) {
+                        $.ajax({
+                            type: "POST",
+                            url: server + "profile",
+                            contentType: false,
+                            processData: false,
+                            data: profileFormData,
+                            success: function(data) {
+                                if(img_hash) {
+                                    $(".topThumb").css("background-image",
+                                        "url(" + server + "get_image?hash=" +
+                                                 img_hash + ")");
+                                    $("#avatar").val("");
+                                }
+                                self.showErrorModal("Saved", "Your settings have been successfully saved");
+                            },
+                            error: function(jqXHR, status, errorThrown){
+                                console.log(jqXHR);
+                                console.log(status);
+                                console.log(errorThrown);
+                                self.showErrorModal("Server error", "Profile API endpoint return an error");
+                            }
+                        });
+                    }
+                },
+                error: function(jqXHR, status, errorThrown){
+                    console.log(jqXHR);
+                    console.log(status);
+                    console.log(errorThrown);
+                    self.showErrorModal("Server error", "Settings API endpoint return an error");
+                }
+            });
+
+        };
+
+        //Lets upload the image first, if there is one
+        //to get the hash
+        if($("#avatar").val()) {
+
+            $.ajax({
+                type: "POST",
+                url: server + "upload_image",
+                contentType: false,
+                processData: false,
+                data: uploadImageFormData,
+                success: function(data) {
+                    submit(JSON.parse(data).image_hashes[0]);
+                },
+                error: function(jqXHR, status, errorThrown){
+                    console.log(jqXHR);
+                    console.log(status);
+                    console.log(errorThrown);
+                    self.showErrorModal("Server error", "Failed to upload image");
+                }
+            });
+
+        } else { //Otherwise lets just submit right away
+            submit();
+        }
+
+  },
+
+  close: function(){
```

Comment:
When you do the merge, take a look at the updates to the close methods we pushed today.